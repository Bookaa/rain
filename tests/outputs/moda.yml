!program;1
stmts:
- !import;1
  name: modb
  rename: null
- !assn;2
  export: true
  let: false
  lhs: !name;1
    value: name
  rhs: !str;1
    value: module a
- !assn;2
  export: true
  let: false
  lhs: !name;1
    value: init
  rhs: !func;1
    body: !block;2
      expr: null
      stmts:
      - !call;1
        args:
        - !str;1
          value: moda init()
        catch: false
        func: !name;1
          value: print
    params: []
- !assn;2
  export: true
  let: false
  lhs: !name;1
    value: test
  rhs: !func;1
    body: !block;2
      expr: null
      stmts:
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'moda.name = '
          op: $
          rhs: !name;1
            value: name
        catch: false
        func: !name;1
          value: print
    params: []
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;2
      expr: null
      stmts:
      - !call;1
        args:
        - !str;1
          value: moda.main()
        catch: false
        func: !name;1
          value: print
      - !call;1
        args: []
        catch: false
        func: !name;1
          value: test
      - !call;1
        args: []
        catch: false
        func: !index;1
          lhs: !name;1
            value: modb
          rhs: !str;1
            value: test
      - !assn;2
        export: false
        let: false
        lhs: !name;1
          value: name
        rhs: !str;1
          value: new a module
      - !assn;2
        export: false
        let: false
        lhs: !index;1
          lhs: !name;1
            value: modb
          rhs: !str;1
            value: name
        rhs: !str;1
          value: new b module
      - !call;1
        args: []
        catch: false
        func: !name;1
          value: test
      - !call;1
        args: []
        catch: false
        func: !index;1
          lhs: !name;1
            value: modb
          rhs: !str;1
            value: test
    params: []
