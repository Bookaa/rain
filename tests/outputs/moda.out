moda init()
modb init()
moda.main()
moda.name = module a
modb.name = module b
moda.name = new a module
modb.name = new b module
